pkg/util/ioutil/ioutil.go

```diff
@@ -25,7 +24,7 @@ func (tr *timeoutReader) Read(p []byte) (int, error) {
 	n, err := tr.r.Read(p)
 
 	switch {
-	case (n == 0 || errors.Is(err, io.EOF)) && tr.t.Stop():
+	case err != nil && tr.t.Stop():
```

Comment:
can we write this function without the `fallthrough`

Reply:
Done.